[inductor] Improve handling of index_expr with floating point dtypes

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107526
* __->__ #105021

I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because triton's `ops.index_expr` expects integer expressions and dtypes.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

peterbell10 committed:
[inductor] Add rounding ops to IndexPropagation pass

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Add rounding ops to IndexPropagation pass"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Add rounding ops to IndexPropagation pass"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Add rounding ops to IndexPropagation pass"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Don't call index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because `ops.index_expr` is intended to only support integer
values.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Don't call index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because `ops.index_expr` is intended to only support integer
values.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Improve handling of index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because triton's `ops.index_expr` expects integer expressions and dtypes.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Improve handling of index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because triton's `ops.index_expr` expects integer expressions and dtypes.

[ghstack-poisoned]

lezcano reviewed (commented):
Nice addition, but could you add a short regression test?

peterbell10 committed:
Update on "[inductor] Improve handling of index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because triton's `ops.index_expr` expects integer expressions and dtypes.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Improve handling of index_expr with floating point dtypes"


I found that the upsample bicubic lowering was generating this line

```python
ops.index_expr(0.244094488188976*x0, torch.float32)
```

which is not good because triton's `ops.index_expr` expects integer expressions and dtypes.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge

peterbell10 commented:
@pytorchbot merge
